### PR TITLE
[IgnoredReturnValue] Add `returnType?.isNothing()` check

### PR DESCRIPTION
**Pull request by atulgpt**
Fixes https://github.com/detekt/detekt/issues/7765

### CODE REVIEW

File: detekt-rules-errorprone/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/IgnoredReturnValueSpec.kt [hunk 1]
```diff
@@ -1056,6 +1056,28 @@ class IgnoredReturnValueSpec {
...
+        @Test
+        fun `does not report when a function returns Unit`() {
+            val code = """
+                fun foo() {
+                    println(42)
+                }
+            """.trimIndent()
+            val findings = subject.compileAndLintWithContext(env, code)
+            assertThat(findings).isEmpty()
+        }
```
Comment (BraisGabin) [inline detekt-rules-errorprone/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/IgnoredReturnValueSpec.kt:1069]: Didn't we have a test for this case? Good catch then!

**Review by BraisGabin — APPROVED** [review]
x"D I was just reviewing the issue and saw "oh there is a PR already". I enter and it was just 1 minute old :P

Thanks for all the contributions (I know that you have more things that need my review but I have not the time :'( )

File: detekt-rules-errorprone/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/IgnoredReturnValueSpec.kt (hunk 1 above)
Comment (atulgpt) [reply detekt-rules-errorprone/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/IgnoredReturnValueSpec.kt:1069]: I could only find Unit involving with chain

**Review by schalkms — COMMENTED** [review]
Thanks for fixing this issue!
I’d also like to have an explicit test case for `Nothing`, e.g.

```kotlin
fun fail(message: String): Nothing {
    throw IllegalArgumentException(message)
}
```

**Review by cortinico — APPROVED** [review]

**Pull request merged**